src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/SelectGitProvider.tsx

```diff
@@ -58,19 +67,36 @@ const SelectGitProviderRef = (
   props: SelectGitProviderProps,
   forwardRef: SelectGitProviderForwardRef
 ): React.ReactElement => {
-  const { selectedGitProvider } = props
+  const { selectedGitProvider, disableNextBtn, enableNextBtn } = props
   const { getString } = useStrings()
-  const [gitProvider, setGitProvider] = useState<GitProvider>()
+  const [gitProvider, setGitProvider] = useState<GitProvider | undefined>(selectedGitProvider)
   const [authMethod, setAuthMethod] = useState<GitAuthenticationMethod>()
   const [testConnectionStatus, setTestConnectionStatus] = useState<TestStatus>(TestStatus.NOT_INITIATED)
   const formikRef = useRef<FormikContext<SelectGitProviderInterface>>()
 
   useEffect(() => {
     if (authMethod === GitAuthenticationMethod.AccessToken) {
       setTestConnectionStatus(TestStatus.NOT_INITIATED)
+      enableNextBtn()
     }
   }, [authMethod])
 
+  useEffect(() => {
+    if (gitProvider) {
+      if (authMethod === GitAuthenticationMethod.AccessToken) {
+        if (testConnectionStatus === TestStatus.SUCCESS) {
+          enableNextBtn()
+        } else {
+          disableNextBtn()
+        }
+      } else {
+        enableNextBtn()
+      }
+    } else {
+      enableNextBtn()
+    }
+  }, [gitProvider, authMethod, testConnectionStatus])
```

Comment:
Codacy has a fix for the issue: [React Hook useEffect has missing dependencies: 'disableNextBtn' and 'enableNextBtn'. Either include them or remove the dependency array.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9254902)
```suggestion
  }, [gitProvider, authMethod, testConnectionStatus, enableNextBtn, disableNextBtn])
```